modules/mobile/spec/request/messages_request_spec.rb

```diff
@@ -0,0 +1,185 @@
+
+      expect(response).to be_successful
+      expect(response.body).to be_a(String)
+      expect(response).to match_camelized_response_schema('category')
+    end
+
+    it 'responds to GET #show' do
+      VCR.use_cassette('sm_client/messages/gets_a_message_with_id') do
+        get "/mobile/v0/messaging/health/messages/#{message_id}", headers: iam_headers
+      end
+
+      expect(response).to be_successful
+      expect(response.body).to be_a(String)
+      expect(response).to match_camelized_response_schema('message')
+    end
+
+    it 'generates mobile-specific metadata links' do
+      VCR.use_cassette('sm_client/messages/gets_a_message_with_id') do
+        get "/mobile/v0/messaging/health/messages/#{message_id}", headers: iam_headers
+      end
+
+      result = JSON.parse(response.body)
+      expect(result['data']['links']['self']).to match(%r{/mobile/v0})
+    end
+
+    describe 'POST create' do
+      let(:attachment_type) { 'image/jpg' }
+      let(:uploads) do
+        [
+          Rack::Test::UploadedFile.new('spec/fixtures/files/sm_file1.jpg', attachment_type),
+          Rack::Test::UploadedFile.new('spec/fixtures/files/sm_file2.jpg', attachment_type),
+          Rack::Test::UploadedFile.new('spec/fixtures/files/sm_file3.jpg', attachment_type),
+          Rack::Test::UploadedFile.new('spec/fixtures/files/sm_file4.jpg', attachment_type)
+        ]
+      end
+      let(:message_params) { attributes_for(:message, subject: 'CI Run', body: 'Continuous Integration') }
+      let(:params) { message_params.slice(:subject, :category, :recipient_id, :body) }
+      let(:params_with_attachments) { { message: params }.merge(uploads: uploads) }
+
+      context 'message' do
+        it 'without attachments' do
+          VCR.use_cassette('sm_client/messages/creates/a_new_message_without_attachments') do
+            post '/mobile/v0/messaging/health/messages', headers: iam_headers, params: { message: params }
+          end
+
+          expect(response).to be_successful
+          expect(response.body).to be_a(String)
+          expect(JSON.parse(response.body)['data']['attributes']['subject']).to eq('CI Run')
+          expect(JSON.parse(response.body)['data']['attributes']['body']).to eq('Continuous Integration')
+          expect(response).to match_camelized_response_schema('message')
+        end
+
+        it 'with attachments' do
+          VCR.use_cassette('sm_client/messages/creates/a_new_message_with_4_attachments') do
+            post '/mobile/v0/messaging/health/messages', headers: iam_headers, params: params_with_attachments
+          end
+
+          expect(response).to be_successful
+          expect(response.body).to be_a(String)
+          expect(JSON.parse(response.body)['data']['attributes']['subject']).to eq('CI Run')
+          expect(JSON.parse(response.body)['data']['attributes']['body']).to eq('Continuous Integration')
+          expect(response).to match_camelized_response_schema('message_with_attachment')
```

Comment:
A little bit painful but for the JSON schema checks we have our own set of _almost_ duplicate schemas dir in support as a way to protect us from the core lib being change from out from under us. At the risk of the PR line count 🤖 coming after you, you could add `match_camelized_response_schema` to our support/matches/json_schema_matcher.rb and copy the core lib ones over. (also helps us avoid needing VSP approval if there's a future mobile only change)

Reply:
That's a good idea; I meant to sync with you about  where you'd landed on duplication vs. re-use, generally.